src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java

```diff
@@ -22,13 +37,74 @@
     private DeleteCommandParser parser = new DeleteCommandParser();
 
     @Test
-    public void parse_validArgs_returnsDeleteCommand() {
-        assertParseSuccess(parser, "1", new DeleteCommand(new Index[]{INDEX_FIRST_PERSON}));
+    public void parse_validIndex_returnsDeleteCommand() {
+        assertParseSuccess(parser, VALID_INDEX, new DeleteCommand(new Index[]{INDEX_FIRST_PERSON}));
     }
 
     @Test
-    public void parse_invalidArgs_throwsParseException() {
-        assertParseFailure(parser, "a", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+    public void parse_multipleValidIndex_returnsDeleteCommand() {
+        assertParseSuccess(parser, VALID_INDEX_MULTIPLE, new DeleteCommand(
+                new Index[]{INDEX_FIRST_PERSON, INDEX_SECOND_PERSON, INDEX_THIRD_PERSON}));
+    }
+
+    @Test
+    public void parse_validStudentId_returnsDeleteCommand() {
+        // valid student id
+        DeleteCommand expectedDeleteCommand = new DeleteCommand(new StudentId(VALID_ID_BOB));
+
+        // no white space before id
+        assertParseSuccess(parser, ID_DESC_BOB, expectedDeleteCommand);
+
+        // whitespace before id
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + ID_DESC_BOB, expectedDeleteCommand);
+    }
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                DeleteCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_nonemptyPreamble_throwsParseException() {
```

Comment:
```suggestion
    public void parse_nonEmptyPreamble_throwsParseException() {
```

Just a minor suggestion for clarity.

Reply:
Thanks for the suggestion, will commit 👍